### ISSUE TITLE
Feature request: auto-allow JS allowed by LibreJS

### ISSUE THREAD

**Issue by strypey** [hackademix/noscript#78]
I would really like an opt-in feature I can use to instruct NoScript to automatically add any Javascript that LibreJS would allow to the 'trusted' list.

**Comment by vecna13** [comment]
My initial reaction to this was, "Why not just use LibreJS?" But this could be useful for e.g., Tor Browser which has NoScript by default and discourages installing new add-ons.

I could see LibreJS rules being useful as a level of trust between "default" and "trusted", as in "this source may not run JS" > "this source may run JS, but only free JS" > "this source may run any JS"

**Comment by strypey** [comment]
> "Why not just use LibreJS?"

Good question :) My use case would be allowing all free JS by default (via an opt-in checkbox in the settings I guess) and blocking all non-free JS by default. So far, so LibreJS.

But then, I would want to be able to selectively trust certain non-free JS (such as internet banking sites and certain government websites) and block any unwanted JS that happens to be free (probably not common now but could be in the future). In some cases I would want to mark things as 'temp trusted' when I need to use them. I admit I haven't used it for a while, but I believe LibreJS only allows domains to be included or excluded in its filter. In other cases, I might want to completely disable restrictions in a tab, to see if it's JS blocking causing a problem with a website. In all these cases, the granular control that NS gives the user makes it much more helpful than LibreJS, at least for now (hopefully it will improve in that respect over time).

**Comment by bill-auger** [comment]
i agree this feature would make noscript and librejs essentially the same program, wouldnt it?

i found this while sorting out the analogous feature requests that have been coming in for librejs to behave more like noscript 

https://github.com/librejs/librejs/issues/1

its difficult to gauge exactly what most people want, but on the surface, it looks like they want some form of a merger of the functionality of both programs

**Comment by vecna13** [comment]
Notably (for Strypey's use case), LibreJS does allow you to whitelist/blacklist individual scripts, but you still can't disable all restrictions on a tab level, and it doesn't seem to have a way to make your changes temporary (besides manually reverting them).
